3rdparty/openssl/CMakeLists.txt

```diff
@@ -868,87 +1466,65 @@ if (WIN32 OR CMAKE_C_COMPILER_ID MATCHES Clang)
-  openssl/ssl/statem/extensions_srvr.c
-  openssl/ssl/statem/statem.c
-  openssl/ssl/statem/statem_clnt.c
-  openssl/ssl/statem/statem_dtls.c
-  openssl/ssl/statem/statem_lib.c
-  openssl/ssl/statem/statem_srvr.c
-  openssl/ssl/t1_enc.c
-  openssl/ssl/t1_lib.c
-  openssl/ssl/t1_trce.c
-  openssl/ssl/tls13_enc.c
-  openssl/ssl/tls_srp.c)
+add_enclave_library(opensslssl STATIC ${OPENSSL_SSL_LIB_SOURCE})
+
+add_enclave_library(opensslssl_3 STATIC ${OPENSSL_3_SSL_LIB_SOURCE})
 
 enclave_include_directories(
   opensslcrypto PRIVATE $<BUILD_INTERFACE:${OPENSSL_BUILD_INCLUDES}>
   $<BUILD_INTERFACE:${OPENSSL_DIR}> $<BUILD_INTERFACE:${OPENSSL_DIR}/include>)
 
+enclave_include_directories(
+  opensslcrypto_3
+  PRIVATE
+  $<BUILD_INTERFACE:${OPENSSL_3_BUILD_INCLUDES}>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}/include>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}/providers/common/include>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}/providers/implementations/include>)
+
 # Only mark the directories that include headers to be published as PUBLIC.
 enclave_include_directories(
   opensslcrypto PUBLIC $<BUILD_INTERFACE:${OE_INCDIR}/openenclave>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty>)
 
+enclave_include_directories(
+  opensslcrypto_3 PUBLIC $<BUILD_INTERFACE:${OE_INCDIR}/openenclave>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty>)
+
 enclave_include_directories(
   opensslssl PRIVATE $<BUILD_INTERFACE:${OPENSSL_BUILD_INCLUDES}>
   $<BUILD_INTERFACE:${OPENSSL_DIR}> $<BUILD_INTERFACE:${OPENSSL_DIR}/include>)
 
+enclave_include_directories(
+  opensslssl_3 PRIVATE $<BUILD_INTERFACE:${OPENSSL_3_BUILD_INCLUDES}>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}>
+  $<BUILD_INTERFACE:${OPENSSL_3_DIR}/include>)
+
 maybe_build_using_clangw(opensslcrypto)
 maybe_build_using_clangw(opensslssl)
+maybe_build_using_clangw(opensslcrypto_3)
+maybe_build_using_clangw(opensslssl_3)
 
 add_enclave_dependencies(opensslcrypto openssl_generated)
 add_enclave_dependencies(opensslssl opensslcrypto)
+add_enclave_dependencies(opensslcrypto_3 openssl_3_generated)
+add_enclave_dependencies(opensslssl_3 opensslcrypto_3)
 
 enclave_link_libraries(opensslcrypto PUBLIC oelibc oe_includes)
 enclave_link_libraries(opensslssl PUBLIC opensslcrypto)
+enclave_link_libraries(opensslcrypto_3 PUBLIC oelibc oe_includes)
+enclave_link_libraries(opensslssl_3 PUBLIC opensslcrypto_3)
 
-list(
-  APPEND
-  OPENSSL_COMPILE_OPTIONS
-  -Wa,--noexecstack
-  -Wall
-  # Suppress the following warnings that are commonly raised during the compilation.
-  -Wno-unused-parameter
-  -Wno-shorten-64-to-32
-  -Wno-sign-conversion
-  -Wno-conversion)
+set(OPENSSL_COMPILE_OPTIONS
+    -Wa,--noexecstack
+    -Wall
+    # Suppress the following warnings that are commonly raised during the compilation.
+    -Wno-unused-parameter
+    -Wno-shorten-64-to-32
+    -Wno-sign-conversion
+    -Wno-conversion
+    -Wno-deprecated-declarations)
```

Comment:
Do we know what cause this warning?

Reply:
We discussed offline, we will plan to remove deprecated API usage in a future change. I don't have a specific list of what causes this warning yet.